ConfigSpace/util.pyx

```diff
@@ -436,3 +436,185 @@ def fix_types(configuration: dict,
+
+    Parameters
+    ----------
+    configuration_space: :class:`~ConfigSpace.configuration_space.ConfigurationSpace`
+        The Configuration space over which to create a grid of HyperParameter Configuration values. It knows the types for all parameter values.
+
+    num_steps_dict: dict
+        A dict containing the number of points to divide the grid side formed by Hyperparameters which are either of type UniformFloatHyperparameter or type UniformIntegerHyperparameter. The keys in the dict should be the names of the corresponding Hyperparameters and the values should be the number of points to divide the grid side formed by the corresponding Hyperparameter in to.
+
+    Returns
+    -------
+    list
+        List containing Configurations. It is a cartesian product of tuples of HyperParameter values. Each tuple lists the possible values taken by the corresponding HyperParameter. Within the cartesian product, in each element, the ordering of HyperParameters is the same for the OrderedDict within the ConfigurationSpace.
+    """
+
+    def get_value_set(num_steps_dict, hp_name):
+        '''
+        Gets values along the grid for a particular hyperparameter.
+
+        Uses the num_steps_dict to determine number of grid values for UniformFloatHyperparameter and UniformIntegerHyperparameter. If these values are not present in num_steps_dict, the quantization, q, objects of these classes will be used to divide the grid. NOTE: If q is None, this results in np.arange using None for its step size parameter which results in a quantization of 1.
+
+        Parameters
+        ----------
+        num_steps_dict: dict
+            Same description as above
+
+        hp_name: str
+            Hyperparameter name
+
+        Returns
+        -------
+        tuple
+            Holds grid values for the given hyperparameter
+
+        '''
+        param = configuration_space.get_hyperparameter(hp_name)
+        if isinstance(param, (CategoricalHyperparameter)):
+            return param.choices
+
+        elif isinstance(param, (OrdinalHyperparameter)):
+            return param.sequence
+
+        elif isinstance(param, Constant):
+            return tuple([param.value, ])
+
+        elif isinstance(param, UniformFloatHyperparameter):
+            if param.log:
+                lower, upper = np.log([param.lower, param.upper])
+            else:
+                lower, upper = param.lower, param.upper
+
+            if param.name in num_steps_dict:
+                num_steps = num_steps_dict[param.name]
+                grid_points = np.linspace(lower, upper, num_steps)
+            else:
+                grid_points = np.arange(lower, upper + param.q, param.q) # check for log and for rounding issues
+
+            if param.log:
+                grid_points = np.exp(grid_points)
+
+            # Avoiding rounding off issues
+            print(grid_points)

```

Comment:
Could you please remove this print call?

Reply:
Done